homeassistant/components/media_player/mpd.py

```diff
@@ -0,0 +1,196 @@
+"""
+homeassistant.components.media_player.mpd
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Provides functionality to interact with a Music Player Daemon.
+
+Configuration:
+
+To use MPD you will need to add something like the following to your
+config/configuration.yaml
+
+media_player:
+  platform: mpd
+  server: 127.0.0.1
+  port: 6600
+  location: bedroom
+
+Variables:
+
+server
+*Required
+IP address of the Music Player Daemon. Example: 192.168.1.32
+
+port
+*Optional
+Port of the Music Player Daemon, defaults to 6600. Example: 6600
+
+location
+*Optional
+Location of your Music Player Daemon.
+"""
+import logging
+import socket
+
+from homeassistant.components.media_player import (
+    MediaPlayerDevice, STATE_NO_APP, ATTR_MEDIA_STATE,
+    ATTR_MEDIA_CONTENT_ID, ATTR_MEDIA_TITLE, ATTR_MEDIA_ARTIST,
+    ATTR_MEDIA_ALBUM, ATTR_MEDIA_DATE, ATTR_MEDIA_DURATION,
+    ATTR_MEDIA_VOLUME, MEDIA_STATE_PAUSED, MEDIA_STATE_PLAYING,
+    MEDIA_STATE_STOPPED, MEDIA_STATE_UNKNOWN)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Sets up the MPD platform. """
+
+    daemon = config.get('server', None)
+    port = config.get('port', 6600)
+    location = config.get('location', 'MPD')
+
+    # pylint: disable=unused-argument
+    try:
+        from mpd import MPDClient
+
+    except ImportError:
+        _LOGGER.exception(
+            "Unable to import mpd2. "
+            "Did you maybe not install the 'python-mpd2' package?")
+
+        return None
+
+    # pylint: disable=unused-argument
+    try:
+        mpd_client = MPDClient()
+        mpd_client.connect(daemon, port)
+    except socket.error:
+        _LOGGER.error(
+            "Unable to connect to MPD. "
+            "Please check your settings")
+
+        return None
+
+    mpd = []
+    if daemon is not None and port is not None:
```

Comment:
The fact that we arrived here should imply that daemon and port are not `None` ?

Also, port will never be `None` as it will use the default value if not found (`6600`).